src/main/kotlin/Main.kt

```diff
@@ -25,59 +27,80 @@ fun main() {
+                            }
+                        }
+
+                        while (true) {
+                            println("Введите текст заметки")
+                            val body = scanner.nextLine()
+                            if (body.isNotEmpty()) {
+                                archiveList[selectedArchiveIndex].notes.add(Note(name, body))
+                                menuState = MenuState.NOTES
+                                break
+                            } else {
+                                println("Вы ввели пустую строку. Попробуйте снова")
+                            }
+                        }
                         continue
                     }
                 }
             }
         }
 
         while (true) {
-            if (scanner.hasNextInt()) {
+            try {
                 val selectedIndex = scanner.nextLine().toInt()
                 menu.select(
-                        selectedIndex,
-                        { menuState = MenuState.CREATE },
-                        { selectedIndex ->
-                            when (menuState) {
-                                MenuState.ARCHIVE -> {
-                                    selectedArchiveIndex = selectedIndex - 1
-                                    menuState = MenuState.NOTES
-                                }
-                                MenuState.NOTES -> {
-                                    selectedNoteIndex = selectedIndex - 1
-                                    menuState = MenuState.VIEW
-                                }
-                                else -> {}
+                    selectedIndex,
+                    { menuState = MenuState.CREATE },
+                    { index ->
+                        when (menuState) {
+                            MenuState.ARCHIVE -> {
+                                selectedArchiveIndex = selectedIndex - 1
+                                menuState = MenuState.NOTES
+                            }
+                            MenuState.NOTES -> {
+                                selectedNoteIndex = selectedIndex - 1
+                                menuState = MenuState.VIEW
                             }
-                        },
-                        { when (menuState) {
-                            MenuState.NOTES -> menuState = MenuState.ARCHIVE
-                            else -> return@select
+                            else -> Unit
                         }
+                    },
+                    { when (menuState) {
+                        MenuState.NOTES -> menuState = MenuState.ARCHIVE
+                        else -> {
+                            scanner.close()
+                            exitProcess(0)
                         }
+                    }
+                    }
                 )
                 break
-            } else {
+            } catch (e: NumberFormatException) {
```

Comment:
🍏 Здесь можно отловить сразу общий Exception, чтобы не допустить падение приложения из-за неверного ввода:
```suggestion
            } catch (e: Exception) {
```